File: solutions/javascript/anagram/1/anagram.js
export const findAnagrams = (target, candidates) => {
  const lowerTarget = target.toLowerCase();
  const sortString = (str) => str.toLowerCase().split('').sort().join('');
  const sortedTarget = sortString(lowerTarget);

  return candidates.filter(candidate => {
    const lowerCandidate = candidate.toLowerCase();
    return lowerCandidate !== lowerTarget && sortString(lowerCandidate) === sortedTarget;
  });
};

File: solutions/julia/anagram/1/anagram.jl
function detect_anagrams(subject::AbstractString, candidates::AbstractVector{<:AbstractString})

    sub_lower = lowercase(subject)
    

    sub_sorted = sort(collect(sub_lower))
    

    filter(candidates) do candidate
        cand_lower = lowercase(candidate)
        
      
        cand_lower != sub_lower && sort(collect(cand_lower)) == sub_sorted
    end
end
